### ISSUE TITLE
Error when using a widget that doesn't generate content

### ISSUE THREAD

**Issue by VadymPogorelov** [elementor/elementor#12509]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

If the page uses a widget that returns empty content (for example, a currency switch that returns content only on store pages), an error occurs: https://www.screencast.com/t/arI3lxD7P

## Steps to reproduce

1. Add any widget that returns empty content (for example, a currency switch that returns content only on store pages) to the page
2. Save the page and update browser (press F5 or Ctrl+F5)
3. An error occurs: https://www.screencast.com/t/arI3lxD7P

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
https://pastebin.com/dxLdgAYN
```
</details>

**Comment by MichaelDarko** [comment]
I have also been experiencing this exact issue since 3.0 came out.

**Comment by VadymPogorelov** [comment]
Yes, the problem appeared in version 3.0. and still remains (the current version is 3.0.8.1)

> 
> 
> I have also been experiencing this exact issue since 3.0 came out.

Yes, the problem appeared in version 3.0. and still remains (the current version is 3.0.8.1)

**Comment by shilo-ey** [comment]
Hi @VadymPogorelov @MichaelDarko 

I'm testing this issue and can't reproduce this issue.
![image](https://user-images.githubusercontent.com/45009641/93225945-74c00200-f77b-11ea-83ca-5c2c795dd003.png)



Have you made sure it's not originating from a 3rd party plugin?

Please try:

- Deactivate all your plugins besides Elementor (and Elementor Pro). If this solves the problem, gradually activate your plugins one by one, until you spot the problematic plugin.
 - Change your WordPress theme to Hello Theme (or other default WordPress theme).


And see if the issue still persists.

Thanks

**Comment by MichaelDarko** [comment]
I have now identified one  case when similar thing happens (But I thing there are other cases.).

- It's happening with widgets that are imported form older 2.x versions. 

- In this case it is a Nav Menu widget on a site with no menus. 

- When you insert new Nav Menu widget, the widget is empy, but it's editable.

- When you import older template with Nav Menu widget, the widget is empy and the widget does not show show the blue "edit" button upon hover.

- When you move the widget to another section, the edit button starts working.

- When you close the editor and open it again, the edit button is gone again.

(Using Hello theme and Elementor, not other plugins.)

Error it generates is here:

```
Uncaught TypeError: Cannot read property 'document' of undefined
    at i.onModelRemoteRender (editor.min.js?ver=3.0.8-20200912.1945:2)
    at p (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:10)
    at f (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:10)
    at l (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:10)
    at i.n.trigger (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:10)
    at i.onRemoteGetHtml (editor.min.js?ver=3.0.8-20200912.1945:2)
    at i (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:4)
    at Object.fireWith [as resolveWith] (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:4)
    at Object.e.<computed> [as resolve] (load-scripts.php?c=0&load[chunk_0]=underscore,jquery-core,backbone,jquery-ui-core,jquery-ui-widget,jquery-ui-mouse,wp-api-request,shortcode,utils,wp-util,wp-backbo&load[chunk_1]=ne,media-models,moxiejs,plupload,wp-plupload,jquery-ui-sortable&ver=5.5.1:4)
    at common.min.js?ver=3.0.8-20200912.1945:2
```

**Comment by VadymPogorelov** [comment]
Widgets that you used for the test in any case return some kind of layout (even if the tags are empty inside-the tags themselves are present): https://www.screencast.com/t/wflbNADFLN4

The error occurs when using widgets that return nothing (return an empty string). In this case, a block is not created for them .elementar-widget-container: https://www.screencast.com/t/A08VFrpnS5Mu
